src/main/javascript/router.js

```diff
@@ -38,7 +46,21 @@ window.SwaggerUiRouter = Backbone.Router.extend({
 
             onFailure: function(data) {
                 if(data === '401 : {\"message\":\"The identity is not set or unauthorized.\"} ' + window.swaggerUi.options.url) {
-                    Backbone.history.navigate('login', true);
+                    var host = window.location;
+                    var pathname = location.pathname.substring(0, location.pathname.lastIndexOf('/'));
+                    var url = host.protocol + '//' + host.host + pathname.replace('swagger', 'login/url');
+                    $.ajax({
+                        url : url,
+                        type: 'POST',
+                        success: function (data)
+                        {
+                            window.location.href = data;
```

Comment:
We are using here full page reload? Or we can use Backbone.history for partial updates only?